frontend/src/components/Bar/Bar.jsx

```diff
@@ -72,69 +73,70 @@ function Bar() {
             </Link>
           </Typography>
         </Box>
-
-        {loggedIn && (
-          <>
-            <Button color="inherit" onClick={openMenu}>
-              <MenuIcon></MenuIcon>
-            </Button>
-
-            <Menu
-              anchorEl={anchorEl}
-              open={Boolean(anchorEl)}
-              onClose={closeMenu}
-            >
-              {menuItems.map((menuItem, index) => {
-                if (
-                  menuItem.hasOwnProperty("condition") &&
-                  !menuItem.condition
-                ) {
-                  return null;
-                }
-
-                let component = null;
-
-                if (menuItem.to) {
-                  component = (
-                    <MenuItem
-                      key={index}
-                      component={RouterLink}
-                      to={menuItem.to}
-                      onClick={closeMenu}
-                    >
-                      {menuItem.name}
-                    </MenuItem>
-                  );
-                } else {
-                  component = (
-                    <MenuItem
-                      key={index}
-                      onClick={() => {
-                        closeMenu();
-
-                        menuItem.onClick();
-                      }}
-                    >
-                      {menuItem.name}
-                    </MenuItem>
-                  );
-                }
-
-                if (menuItem.divide) {
-                  return (
-                    <span key={index}>
-                      <Divider />
-
-                      {component}
-                    </span>
-                  );
-                }
-
-                return component;
-              })}
-            </Menu>
-          </>
-        )}
+        {/* The filter removes all elements that are "true" or "false" */}
+        {loggedIn &&
+          menuItems.filter((e) => typeof e !== "boolean").length > 0 && (
```

Comment:
I'm trying to figure out the meaning of this filter. Could you explain, please?

Reply:
In lines 50-54, we are adding the "log out" button if `disableAuth` is set to false. When that happens, the array will have a value of `false` instead. The reason for the check of the length is to remove the hamburger menu if there are no elements to be shown. Because the array will have a "false" in it instead of an object, it still counts as an element. This filter removes all boolean values to check the length.

If there's a better way to conditionally add an element to a list without messing with const/whatever, I would be glad to change it.

Reply:
That seems a better approach: https://2ality.com/2017/04/conditional-literal-entries.html

Reply:
Applied!